Provide possibility to use every option available in jira-client

EugenEistrachKC opened the pull request:
There are certain cases were one might need to set some jira-client option that is not part of the config api that jira-changelog exposes. 
The company I work in is using a self hosted jira instance which does not support adding API token and username password auth does not work anymore either for me. Only option I had left was creating personal token and using the "bearer" option of jira-client. 
Thats the main reason for this change but I think it would still be good to provide people little bit more control about the configuration if they are forced to use something specific

jgillick commented on src/Jira.js:
```diff
@@ -22,7 +22,7 @@ export default class Jira {
 
     const { host, username, password} = config.jira.api;
-    let { email, token } = config.jira.api;
+    let { email, token, originalOptions } = config.jira.api;
```
Let's just call it `options` for carity.

EugenEistrachKC replied:
@jgillick please check if that's fine now

jgillick reviewed (changes requested):
Thanks for your contribution. For brevity/clarity, let's just call that config property `options`. Please also add something in the README doc so others know they can use this.

EugenEistrachKC commented:
@jgillick I have implemented requested changes. Also after thinking little bit more I have adjusted the order of properties to allow overwriting hardcoded values like protocol. Only hardcoded property I left was api version. 
Please check if that's fine or if you prefer something else

jgillick commented:
This addition has been published with version 2.1.0 on npm.
